src/lib.rs

```diff
@@ -517,6 +517,91 @@ impl<T: PrimInt + Unsigned> SparseSet<T> {
         }
     }
 
+    /// Return `true` if `self` (A) is an improper subset of `other` (B):
+    ///
+    /// A ⊆ B.
+    ///
+    /// This is the same as [`Self::is_superset`] with the arguments flipped.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use thinset::{set, SparseSet};
+    ///
+    /// let a: SparseSet<u64> = set![6, 3, 5, 64, 2];
+    /// let b = {
+    ///     let mut b = a.clone();
+    ///     b.insert(14);
+    ///     b
+    /// };
+    ///
+    /// // `b` contains all elements that `a` contains.
+    /// assert!(a.is_subset(&b) && b.is_superset(&a));
+    /// // `a` != `b`, so `a` is a proper subset of `b`.
+    /// assert!(a.is_proper_subset(&b));
+    /// ```
+    pub fn is_subset(&self, other: &Self) -> bool {
+        for x in self.iter() {
+            if !other.contains(x) {
+                return false;
+            }
+        }
+
+        true
+    }
+
+    /// Return `true` if `self` (A) is an improper superset of `other` (B):
+    ///
+    /// A ⊇ B.
+    ///
+    /// This is the same as [`Self::is_subset`] with the arguments flipped.
+    pub fn is_superset(&self, other: &Self) -> bool {
+        for x in other.iter() {
+            if !self.contains(x) {
+                return false;
+            }
+        }
+
+        true
+    }
+
+    /// Return `true` if `self` (A) is a proper subset of `other` (B):
+    ///
+    /// A ⊊ B.
+    ///
+    /// This is the same as [`Self::is_proper_superset`] with the arguments flipped.
+    pub fn is_proper_subset(&self, other: &Self) -> bool {
+        self.is_subset(other) && self.inner.len() < other.inner.len()
```

Comment:
Would it be faster to perform the size check first due to boolean short circuiting?

(in the case the result is `true`, both sides have to be computed so the order doesn't matter -- but in the case it's `false`, you'd rather perform the cheaper computation first)
```suggestion
        self.inner.len() < other.inner.len() && self.is_subset(other)
```

Reply:
Yes, it would! I'll update both cases